multiraft/multiraft_test.go

```diff
@@ -392,6 +392,70 @@ func TestMembershipChange(t *testing.T) {
 		}*/
 }
 
+func TestRemoveLeader(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	stopper := stop.NewStopper()
+	cluster := newTestCluster(nil, 6, stopper, t)
+	defer stopper.Stop()
+
+	// Consume and apply the membership change events.
+	for i := 0; i < 4; i++ {
+		go func(i int) {
+			for {
+				if e, ok := <-cluster.events[i].MembershipChangeCommitted; ok {
+					e.Callback(nil)
+				} else {
+					return
+				}
+			}
+		}(i)
+	}
+
+	// Tick all the clocks in the background to ensure that all the
+	// necessary elections are triggered.
+	// TODO(bdarnell): newTestCluster should have an option to use a
+	// real clock instead of a manual one.
+	stopper.RunWorker(func() {
+		ticker := time.NewTicker(10 * time.Millisecond)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-stopper.ShouldStop():
+				return
+			case <-ticker.C:
+				for _, t := range cluster.tickers {
+					t.Tick()
+				}
+			}
+		}
+	})
+
+	// Create a group with three members.
+	groupID := proto.RangeID(1)
+	cluster.createGroup(groupID, 0, 3)
```

Comment:
Three is the minimum sensible group size. This test starts a six-node cluster and migrates a group from the first three nodes to the last three. I'll add an explanatory comment.